packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts

```diff
@@ -0,0 +1,98 @@
+import { ArnFormat, Lazy, Names, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnInsightRule } from './cloudwatch.generated';
+
+/**
+ * Valid states for the rule.
+ */
+export enum RuleState {
+  /**
+   * Insight rule is on
+   */
+  ENABLED = 'ENABLED',
+
+  /**
+   * Insight rule is off
+   */
+  DISABLED = 'DISABLED',
+}
+
+/**
+ * Propperties of defining a CloudWatch Contributor Insights rule.
+ */
+export interface InsightRuleProps {
+  /**
+   * The definition of the rule, as a JSON object.
+   */
+  readonly ruleBody: string;
+
+  /**
+   * The name of the rule
+   *
+   * Up to 128 letters (uppercase and lowercase), numbers, hyphens, and underscores are allows.
+   *
+   * @default - automatically generated name
+   */
+  readonly insightRuleName?: string;
+
+  /**
+   * The current state of the rule
+   *
+   * @default - ENABLED
```

Comment:
```suggestion
   * @default RuleState.ENABLED
```

The `-` is for when we're using plain english to describe the default, as it lets the docstring generator understand what to highlight.